apps/web/lib/thread-hooks.ts

```diff
@@ -101,23 +101,61 @@ export function usePositioning(
   return { isLeftPanel, historyPosition };
 }
 
+/**
+ * Preprocesses content to convert @ComponentName mentions to HTML spans
+ * that can be styled as mention badges. Only converts mentions that match
+ * actual interactable component names.
+ * @param content - The content string to process
+ * @param interactableNames - Optional list of interactable component names to match against
+ * @returns Processed content string with mentions converted to HTML spans
+ */
+function preprocessMentions(
+  content: string,
+  interactableNames?: string[],
+): string {
+  // If no interactable names provided, don't process mentions
+  // (fallback to plain text to avoid false positives)
+  if (!interactableNames || interactableNames.length === 0) {
+    return content;
+  }
+
+  // Create a case-insensitive set for fast lookup
+  const nameSet = new Set(
+    interactableNames.map((name) => name.toLocaleLowerCase()),
+  );
+
+  // Match @ComponentName patterns and only convert if they match an interactable
+  return content.replace(/@([A-Za-z][A-Za-z0-9]*)/g, (match, componentName) => {
+    // Check if this component name matches any interactable (case-insensitive)
+    if (nameSet.has(componentName.toLocaleLowerCase())) {
+      return `<span class="mention-badge-inline">${match}</span>`;
```

Comment:
I don't think we should be doing case-insensitive matches here - we can do insentitive matches in the popup, but here it should be exact